Modify login test file

giosos opened the pull request:
Modify log in test file and add social network buttons connectivity tests. Add in log in with valid creds test holding with mouse and logout parameters.

Add in Homepage file logout button locator, click on logout button method hold cursor on user icon method

Add log in page file locators for Google, Facebook an apple. Add methods for Google, Facebook and apple. Add locators for check box on pop up window for terms & conditions. Add method for click to connect button in pop up window for terms & conditions

nataliaZham committed:
adding ChangeFirstAndLastName Test

nataliaZham committed:
move common locators to Page

nataliaZham committed:
edit variable names and test name

nataliaZham committed:
adding Faker to the "change data" method and remove the unnecessary method from Profile Page

nataliaZham committed:
fixing mistakes in all classes

nataliaZham committed:
clean imports

giosos committed:
Modify log in test file and add social network buttons connectivity tests. Add in log in with valid creds test holding with mouse and logout parameters.

Add in Homepage file logout button locator, click on logout button method hold cursor on user icon method

Add log in page file locators for Google, facebook an apple. Add methods for Google, facebook and apple. Add locators for check box on pop up window for terms & conditions. Add method for click to connect button in pop up window for terms & conditions

giosos committed:
Modify log in test file and add social network buttons connectivity tests. Add in log in with valid creds test holding with mouse and logout parameters.

Add in Homepage file logout button locator, click on logout button method hold cursor on user icon method

Add log in page file locators for Google, facebook an apple. Add methods for Google, facebook and apple. Add locators for check box on pop up window for terms & conditions. Add method for click to connect button in pop up window for terms & conditions

nataliaZham commented on src/test/java/selenide/LoginTest.java:
```diff
@@ -14,15 +14,54 @@ public class LoginTest {
         homePage.accCookies();
         loginPage = homePage.clickOnLoginBtn();
         loginPage.fillInValidCred();
         loginPage.clickOnLoginBtnHP();
```
не хватает в шагах связи между loginPage и  homePage. То есть на 25 строке должно быть homePage = loginPage.clickOnLoginBtnHP();

violettka commented on src/test/java/selenide/LoginTest.java:
```diff
@@ -14,15 +14,54 @@ public class LoginTest {
      */
     @Test
-    public void loginWithValidCredentials() {
+    public void loginWithValidCredentials() throws InterruptedException {
```
Зачем тут обработка  Exception?

giosos replied:
На самом деле не надо!
Я поставил sleep, хотел что-то проверить, а sleep импортирует exception! 
Я потом sleep убрал, но забил убрать exception!

violettka commented on src/test/java/selenide/LoginTest.java:
```diff
@@ -14,15 +14,54 @@ public class LoginTest {
         loginPage.clickOnLoginBtnHP();
         homePage.userIconPresent();
+        homePage.userIconHover();
+        homePage.clickOnLogoutBtn();
```
Я бы не меняла этот тест. Логин тест отдельно - это системный тест, он нужен. Если вы хотите сделать логин-логаут связку - то нужно новый тест создавать.

violettka replied:
Не вижу проверки успешного логаута

violettka commented on src/main/java/selenide/pages/Homepage.java:
```diff
@@ -30,4 +31,12 @@ public LoginPage clickOnLoginBtn() {
+        $(userIcon).hover();
+    }
+
+    public void clickOnLogoutBtn(){
```
здесь он должен возвращать класс главной страницы или той, на которую юзер переходит после логаута

violettka commented on src/main/java/selenide/pages/LoginPage.java:
```diff
@@ -10,18 +10,61 @@ public class LoginPage extends Page {
+        $(connectBtn).click();
+
+        }
+    public void checkTermsAppleBtn(){
```
Где тут проверка в методах check?

violettka commented on src/main/java/selenide/pages/LoginPage.java:
```diff
@@ -10,18 +10,61 @@ public class LoginPage extends Page {
+        $(facebookBtn).click();
+    }
+
+    public void appleBtn(){
```
нужно переделать название метода. clickAppleBtn() и другие тоже - не понятно, что этот метод делает, пока не зайти и не посмотреть.

giosos replied:
Уже сделано )))

nataliaZham committed:
change test

violettka committed:
Merge pull request #3 from violettka/Natalia

adding ChangeFirstAndLastName Test

nataliaZham commented on src/test/java/selenide/LoginTest.java:
```diff
@@ -14,15 +14,54 @@ public class LoginTest {
+    @Test
+    public void googleBtnValidation() {
+        homePage = open(BASE_URL, Homepage.class);
+        loginPage = homePage.clickOnLoginBtn();
```
после открытия homepage не хватает метода accCookies,у меня из-за этого упал тест, не успев начаться :)

giosos replied:
Правильное замечание!
Дело в том что если отдельно запустить тесты для google, apple, facebook, подают, потому что у них не прописан отдельный accCookies! Это потому что в методе loginWithValidCredentials уже существует accCookies!  Если запустить тест (главный) на 10 строке все тесты пройдут! А если прописать отдельный accCookies для google, apple, facebook тогда будет подать главный тест! Согласен, надо доработать!

nataliaZham replied:
Аа,поняла вас,спасибо.
 Но почему ваши тесты будут падать,если в них прописать метод accCookies? Я так и сделала в прошлый раз,чтобы посмотреть ваш тест доконца,у меня все сработало и ничего не упало!

nataliaZham replied:
аа,я поняла,почему остальные тесты будут падать - потому что после главного теста кукис перестанет запрашиваться.
В таком случае можно сделать метод с if else для кукис (это мне Виолетта советовала в прошлый раз)
И таким образом если кукис будут появляться,то на них будет срабатывать метод,а если не появятся,то тест не упадет потому что будет if else)

nataliaZham commented on src/test/java/selenide/LoginTest.java:
```diff
@@ -14,15 +14,54 @@ public class LoginTest {
+    @Test
+    public void facebookBtnValidation(){
+        homePage = open(BASE_URL, Homepage.class);
+        loginPage = homePage.clickOnLoginBtn();
```
тоже самое по поводу метода accCookies

giosos replied:
То же самое что с goolge

nataliaZham commented on src/test/java/selenide/LoginTest.java:
```diff
@@ -14,15 +14,54 @@ public class LoginTest {
+     */
+    @Test
+    public void appleBtnValidation(){
+        homePage = open(BASE_URL, Homepage.class);
```
и снова метод accCookies

giosos replied:
То же самое что с goolge

nataliaZham commented on src/test/java/selenide/LoginTest.java:
```diff
@@ -14,15 +14,54 @@ public class LoginTest {
+        homePage = open(BASE_URL, Homepage.class);
+        loginPage = homePage.clickOnLoginBtn();
+        loginPage.googleBtn();
+        loginPage.checkTermsGoogleBtn();
```
на мой взгляд не хватает дальнейшего действия,а именно - перехода на страницу,в которой нужно выбрать свой Гугл аккаунт. И там хочется сделать проверку на то,что действительно есть возможность выбора аккаунта. Например,взять локатор текста "Выберите аккаунт" и сделать  с ним метод should have text.
А то сейчас тест  заканчивается просто кликом на кнопку connect и все,а подтверждения нет, что после этого открывается страница с выбором аккаунта.

giosos replied:
Переход на странице есть, открывается страница google где можно выбрать account, но стразу закрывается, и нет подтверждение этого я согласен!
Надо доработать!

nataliaZham replied:
Да,переход есть действительно,я неправильно выразилась,имела ввиду,что не хватает финальной проверки ))

nataliaZham commented on src/main/java/selenide/pages/LoginPage.java:
```diff
@@ -10,18 +10,61 @@ public class LoginPage extends Page {
+    private static By appleBtn = By.xpath("//*[@data-qa='social-provider-btn-apple']");
+
+
+    private static By checkTermsGoogle = By.cssSelector("#root>div>div>form>section>div:nth-child(2)>div.consentModal__Wrapper-sc-1favlfo-0.fJWyJT>label>i>svg");
```
19,20,21 строчки содержат один и тот же локатор,есть смысл убрать два и оставить только один,назвав его просто "checkTermsBtn". 
И сам локатор очень длинноватый,если хотите,то можно заменить на //*[@class='Svg-jb9ekn-0 bZETVP nc-icon outline'] - он покороче

giosos replied:
Да согласен, что длинный! Дело в том что у google, apple, facebook у каждого свои "Terms", невозможно использовать один для всех! Обратите внимание на локаторы там внутри записан   :nth-child(2),   :nth-child(3),   :nth-child(4)
A //*[@Class='Svg-jb9ekn-0 bZETVP nc-icon outline'] не подходит!

nataliaZham replied:
да,вы правы

nataliaZham commented on src/main/java/selenide/pages/LoginPage.java:
```diff
@@ -10,18 +10,61 @@ public class LoginPage extends Page {
     LoginPage loginPage;
```
здесь не нужно обьявлять эту переменную, она нужна только в том классе,где написан тест,чтобы к ней обращаться)

giosos replied:
Да я согласен! 
Когда я импортировал repository там уже был это!
Как я понимаю у всех это осталось если не удалили!

giosos committed:
Update 08.10.2022
1. Modify accept cookies method. Now clicks if cookies button exists!
2. Add in Google, Facebook and Apple tests click on accept cookies!
3. Complete tests for social network buttons

giosos committed:
Update 08.10.2022
1. Modify accept cookies method. Now clicks if cookies button exists!
2. Add in Google, Facebook and Apple tests click on accept cookies!
3. Complete tests for social network buttons

giosos committed:
Update 08.10.2022
1. Modify accept cookies method. Now clicks if cookies button exists!
2. Add in Google, Facebook and Apple tests click on accept cookies!
3. Complete tests for social network buttons

giosos committed:
Update 08.10.2022
1. Modify accept cookies method. Now clicks if cookies button exists!
2. Add in Google, Facebook and Apple tests click on accept cookies!
3. Complete tests for social network buttons

giosos committed:
Merge remote-tracking branch 'origin/Gio' into Gio

# Conflicts:
#	src/main/java/selenide/pages/Homepage.java
#	src/main/java/selenide/pages/Page.java

giosos committed:
1. Complete tests in login test file.
2. Add documentation for social network locators.
3. Make joint method for checking redirection on social network websites (Google, Facebook and Apple)

giosos committed:
1. Complete tests in login test file.
2. Add documentation for social network locators.
3. Make joint method for checking redirection on social network websites (Google, Facebook and Apple)

violettka commented on src/main/java/selenide/pages/ProfilePage.java:
```diff
@@ -0,0 +1,27 @@
+    private static final By firstNameField = By.xpath("//*[@data-qa='edit-first-name']");
+    private static final By lastNameField = By.xpath("//*[@data-qa='edit-last-name']");
+    private static final By saveChangesBtn = By.xpath("//*[@data-qa='edit-profile-btn']");
+    private static final By userReservationsBtn = By.xpath("//*[@data-qa='navigation-reservations']");
```
этот локатор используется?

giosos replied:
Это не мой )))

violettka commented on src/main/java/selenide/pages/Page.java:
```diff
@@ -17,14 +23,47 @@ public abstract class Page {
+        }
+
+
+
```
убери лишние линии и отформатируй файлы, пожалуйста

giosos replied:
Убрал

violettka commented on src/main/java/selenide/pages/Page.java:
```diff
@@ -17,14 +23,47 @@ public abstract class Page {
+
+
+
+    public void accCookiesFB() {
```
мы сюда помещаем методы, которые теоретически могут использоваться другими классами. Этот относится к Фейсбуку и логину? Может, лучше туда его перенести?

giosos replied:
Это относиться только Facebook, потому что когда попадаем на станице Facebook, сразу просит accept cookies. Да можно и в Login перенести

violettka commented on src/main/java/selenide/pages/Homepage.java:
```diff
@@ -30,4 +31,24 @@ public LoginPage clickOnLoginBtn() {
+        $(userIcon).hover();
+    }
+
+    public void clickOnLogoutBtn() {
```
я бы перенесла этот метод в Page. Можно выйти из аккаунта на любой странице

giosos replied:
Да, хорошая идея

giosos committed:
Remove unnecessary files

giosos committed:
correct files
